commands/ls_print.go

```diff
@@ -0,0 +1,206 @@
+package commands
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/docker/buildx/builder"
+	"github.com/docker/buildx/store"
+	"github.com/docker/buildx/util/platformutil"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/command/formatter"
+)
+
+const (
+	lsNameNodeHeader       = "NAME/NODE"
+	lsDriverEndpointHeader = "DRIVER/ENDPOINT"
+	lsStatusHeader         = "STATUS"
+	lsLastActivityHeader   = "LAST ACTIVITY"
+	lsBuildkitHeader       = "BUILDKIT"
+	lsPlatformsHeader      = "PLATFORMS"
+
+	lsIndent = ` \_ `
+
+	lsDefaultTableFormat = "table {{.NameNode}}\t{{.DriverEndpoint}}\t{{.Status}}\t{{.Buildkit}}\t{{.Platforms}}"
+	lsDefaultRawFormat   = `{{- if not .IsNode -}}
+name: {{.NameNode}}
+{{- if .DriverEndpoint }}
+driver: {{.DriverEndpoint}}
+{{- end }}
+{{- if .LastActivity }}
+last_activity: {{.LastActivity}}
+{{- end }}
+{{- if .Error }}
+error: {{.Error}}
+{{- else }}
+nodes:
+{{- end }}
+{{- else }}  name: {{.NameNode}}
+  endpoint: {{.DriverEndpoint}}
+  status: {{.Status}}
+  {{- if .Buildkit }}
+  buildkit: {{.Buildkit}}
+  {{- end }}
+  {{- if .Platforms }}
+  platforms: {{.Platforms}}
+  {{- end }}
+  {{- if .Error }}
+  error: {{.Error}}
+  {{- end }}
+{{- end }}
+`
+)
+
+func lsPrint(dockerCli command.Cli, current *store.NodeGroup, builders []*builder.Builder, format string) (hasErrors bool, _ error) {
+	switch format {
+	case formatter.TableFormatKey:
+		format = lsDefaultTableFormat
+	case formatter.RawFormatKey:
+		format = lsDefaultRawFormat
```

Comment:
TBH, I'm always a bit on the fence on the `raw` format. I know we have it for (some?) object-types, but wonder how much it's used, and if there are specific use-cases for it, because we already have `json` for a "parseable" format, and the regular output and `table` for other formats.

Reply:
I see we are using the `raw` format for some commands already in cli: https://github.com/search?q=repo%3Adocker%2Fcli+path%3Acli%2Fcommand+RawFormatKey&type=code&ref=advsearch

Reply:
Yeah, I know it's there, but I _really_ doubt it's used much, and really doubt if anyone would miss it if it's not there (yet).

I can see it somewhat more useful for single objects, but in this case, it's neither structured, not (really) usable for automation or anything.

e.g., I  can't just grep for, well, anything really (because there's structured into, and I can't use it in anyway structured), and it's YAML(ish), but not usable as that either.

(fun fact: know that some people managed to parse the default `docker info` output as YAML (yes, apparently that worked!).

So if it's not targeting "parsing" / "scripting", then it should be targeting "humans", and (honestly), I don't think `raw` output is pretty / readable enough for Humans.

So "tempting" to consider "leave it out" (for now), as it'd be easy to add later if people _really_ _really_ miss it (whereas "remove later" is a lot more difficult)

Reply:
Sure I'm fine removing it in the meantime :+1: